scripts/jest/jest-setup.ts

```diff
@@ -2,6 +2,20 @@
 import '@testing-library/jest-dom';
 import util from 'util';
 
+Object.defineProperty(window, 'matchMedia', {
+    writable: true,
+    value: jest.fn().mockImplementation(query => ({
+        matches: false,
+        media: query,
+        onchange: null,
+        addListener: jest.fn(), // Deprecated
+        removeListener: jest.fn(), // Deprecated
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+    })),
+});
+
```

Comment:
resolve issue TypeError: window.matchMedia is not a function when running tests
https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom

Reply:
if i recall this occurs when using `fireEvent`, can you try using `userEvent` for tests?

Reply:
I think he has to use `fireEvent.change`, I didnt see a way around it when I was doing something similar in the other PR, for everything else he could probably swap out with `userEvent`

Reply:
I swap out all fireEvent with userEvents now but still face this error. The matchMedia function might be wrapped internally in blueprint's modal.

@greg-in-a-box for `fireEvent.change()`, we could use `userEvent.clear(textInput)` + `userEvent.type(textInput, 'xxx')` to replace.